src/devices/ServoMotor/ServoMotor.cs

```diff
@@ -0,0 +1,133 @@
+        /// <param name="minimumPulseWidthMicroseconds">The minimum pulse width, in microseconds, that represent an angle for 0 degrees.</param>
+        /// <param name="maximumPulseWidthMicroseconds">The maxnimum pulse width, in microseconds, that represent an angle for maximum angle.</param>
+        public ServoMotor(
+            int chip,
+            int channel,
+            int maximumAngle = 180,
+            int minimumPulseWidthMicroseconds = 1_000,
+            int maximumPulseWidthMicroseconds = 2_000)
+            : this(maximumAngle, minimumPulseWidthMicroseconds, maximumPulseWidthMicroseconds)
+        {
+            _pwmChannel = PwmChannel.Create(chip, channel, 50);  // Default 50Hz
+        }
+
+        private static double CalculateServoAngleSlope(int maximumAngle, int minimumPulseWidthMicroseconds, int maximumPulseWidthMicroseconds)
+        {
+            if (maximumAngle < 0 || maximumAngle > 360)
+            {
+                throw new ArgumentOutOfRangeException(nameof(maximumAngle), maximumAngle, "Value must be between 0 and 360.");
+            }
+
+            if (minimumPulseWidthMicroseconds < 0)
+            {
+                throw new ArgumentOutOfRangeException(nameof(minimumPulseWidthMicroseconds), minimumPulseWidthMicroseconds, "Value must not be negative.");
+            }
+
+            if (maximumPulseWidthMicroseconds < minimumPulseWidthMicroseconds)
+            {
+                throw new ArgumentOutOfRangeException(nameof(maximumPulseWidthMicroseconds), maximumPulseWidthMicroseconds, $"Value must be greater than or equal to {minimumPulseWidthMicroseconds}.");
+            }
+
+            // slope = (y2 - y1) / (x2 - x1).  x1 is 0 in this case, so it is not subtracted.
+            double servoAngleSlope = (double)(maximumPulseWidthMicroseconds - minimumPulseWidthMicroseconds) / maximumAngle;
+            return servoAngleSlope;
+        }
+
+        /// <summary>
+        /// Starts the servo motor.
+        /// </summary>
+        public void Start() => _pwmChannel.Start();
+
+        /// <summary>
+        /// Stops the servo motor.
+        /// </summary>
+        public void Stop() => _pwmChannel.Stop();
+
+        /// <summary>
+        /// Writes an angle to the servo motor.
+        /// </summary>
+        /// <param name="angle">The angle to write to the servo motor.</param>
+        public void WriteAngle(int angle)
+        {
+            if (angle < 0 || angle > _maximumAngle)
+            {
+                throw new ArgumentOutOfRangeException(nameof(angle), angle, $"Value must be between 0 and {_maximumAngle}.");
+            }
+
+            WritePulseWidth((int)(_servoAngleSlope * angle + _minimumPulseWidthMicroseconds));
+        }
+
+        /// <summary>
+        /// Writes a pulse width to the servo motor.
+        /// </summary>
+        /// <param name="microseconds">The pulse width, in microseconds, to write to the servo motor.</param>
```

Comment:
I feel like I don't understand this description even after reading the code - you seem to be changing fill of the PWM, not really writing anything, also not sure why is this public

Reply:
I might be missing how this device works, are you settting an angle by setting PWM fill? if that's the case I'd call it `SetAngle`

Reply:
ok, I think I understand why this is public (calibration) - other part of the comment stays

Reply:
> if that's the case I'd call it `SetAngle` or `SetDesiredAngle`

Yes, this is basically setting the active pulse width for signal.  You change this to change the position of motor.

I thought about using  `Set`, but thought `Write` was more appropriate related to the #626 discussions going on.

To me, `Set`/`Get` should be used for property setters/getters (at least when accessing non-public/writable things) and `Write`/`Read` should be used when updating binding stuff over the wire.  This can be updated based on #626 discussion results.

Reply:
@shaggygi the difference here is that in control engineering `set position` refers to desired angle. Write suggest you transmit some data somewhere which is not the case here. For sensors you usually read the data from the sensor or write some data to a sensor - in here we literally want to set a position of the servo and hence I think it should have Set in the name. Since the action actually starts the movement and not immediately sets the position (even though it might seem it is immediate) I'd prefer to have some words suggesting that you are about to start the movement into a desired position. PulseWidth basically doesn't mean anything for users who don't know internals on how servo works while SetDesiredAngle has correlation with physical properties which are immediately obvious